GameServer.cpp

```diff
@@ -555,11 +555,10 @@ unsigned int GameServer::GetNewRandValue()
 
 int GameServer::ReceiveFrom(Packet &packet, sockaddr_in &from)
 {
-	int numBytes;
 	socklen_t addrLen = sizeof(from);
 
 	// Read any received packets
-	numBytes = recvfrom(hostSocket, (char*)&packet, sizeof(packet), 0, (sockaddr*)&from, &addrLen);
+	int numBytes = recvfrom(hostSocket, (char*)&packet, sizeof(packet), 0, (sockaddr*)&from, &addrLen);

```

Comment:
Sadly, it seems like using an API with an `unsigned` return value would have been better here, where `SOCKET_ERROR` was set using `std::numeric_limits<T>::max()`, but that's not really in our power to change. That means we're stuck trying to match documentation.

You're right this needs to be updated.

For reference:
Windows: [`recvfrom`](https://docs.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-recvfrom)
Linux: [`recvfrom`](https://linux.die.net/man/2/recvfrom)

It seems Windows uses `int`, while Linux uses `ssize_t`. In both cases, a signed value is returned, though the size of the type may differ.

Perhaps we could make use of `auto` here, so the size would match that of the platform.

----

GCC (versions 7 and 8) produces a warning with the new code:
```
GameServer.cpp:581:15: warning: comparison of integer expressions of different signedness: ‘int’ and ‘long unsigned int’ [-Wsign-compare]
  if (numBytes < sizeof(packet.header) + sizeof(packet.tlMessage.tlHeader))
      ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
GameServer.cpp:587:60: warning: comparison of integer expressions of different signedness: ‘long unsigned int’ and ‘int’ [-Wsign-compare]
  if ((packet.header.sizeOfPayload + sizeof(packet.header)) != numBytes)
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
```

So although the return type might really be signed, and the comparison with `SOCKET_ERROR` uses a signed type, the comparison with the result of `sizeof` is unsigned. I think that was one of the reasons why I never cleaned it up before. We're using the variable in both a signed and an unsigned context.

I'm not sure how best to advise here. Perhaps a check for negative values that might abort by returning `PacketNone`, and otherwise using a `static_cast` to an unsigned value?